piptools/_compat/pip_compat.py

```diff
@@ -3,18 +3,18 @@
 
 import pip
 from pip._internal.index.package_finder import PackageFinder
-from pip._internal.network.session import PipSession
 from pip._internal.req import InstallRequirement
 from pip._internal.req import parse_requirements as _parse_requirements
 from pip._internal.req.constructors import install_req_from_parsed_requirement
 from pip._vendor.packaging.version import parse as parse_version
+from pip._vendor.requests.sessions import Session
 
 PIP_VERSION = tuple(map(int, parse_version(pip.__version__).base_version.split(".")))
 
 
 def parse_requirements(
     filename: str,
-    session: PipSession,
```

Comment:
Oh PipSession/Session is from pyright marking it as a type error. I can undo, but as I was working on this did want to mention pyright detects several type errors that mypy misses. I'd estimate like 1-10 type errors per file.

I think most type errors are inconsistencies in some of the pip types used. Like InstallRequirement vs InstallCandidate in find_best_match. The return type is inconsistent with actual reveal type value and while I'm guessing the two are intended to be compatible they are not currently. Here PipSession is used but the function called works with a more generic Session and one of the callers had more generic type.